actionview/lib/action_view/helpers/form_with_helper.rb

```diff
@@ -0,0 +1,196 @@
+module ActionView
+  module Helpers
+    module FormWithHelper
+      class FormWithBuilder
+        include ActionView::Helpers::TagHelper
+
+        FIELD_HELPERS = [:fields_for, :label, :check_box, :radio_button,
+                        :hidden_field, :file_field, :text_field,
+                        :password_field, :color_field, :search_field,
+                        :telephone_field, :phone_field, :date_field,
+                        :time_field, :datetime_field, :datetime_local_field,
+                        :month_field, :week_field, :url_field, :email_field,
+                        :number_field, :range_field]
+
+        GENERATED_FIELD_HELPERS = FIELD_HELPERS - [:label, :check_box, :radio_button, :fields_for, :file_field, :text_area]
+
+        attr_accessor :model, :scope, :url, :remote
+
+        GENERATED_FIELD_HELPERS.each do |selector|
+          class_eval <<-RUBY_EVAL, __FILE__, __LINE__ + 1
+            def #{selector}(method, *args, **options)                             # def text_field(method, *args, **options)
+              tag.input options_for_field('#{selector}', method, args, options)   #   tag.input options_for_field(selector, method, args, options)
```

Comment:
Not sure what you mean. Do you meant this string: `'#{selector}'`? It will evaluate during class loading. The generated method will look like this (after including feeback from the other comment; tested in console):

```
def text_field(attribute, *args, **options)                                              
  tag.input field_options('text', attribute, *args, options) 
end

```